client/src/app/site/base/base-filter.service/base-filter-list.service.ts

```diff
@@ -382,6 +382,32 @@ export abstract class BaseFilterListService<V extends BaseViewModel> implements
      */
     protected abstract getFilterDefinitions(): OsFilter<V>[];
 
+    private async loadFilters(): Promise<OsFilter<V>[] | null> {
+        return this.parseFilters(await this.activeFiltersStore.load<V>(this.storageKey));
+    }
+
+    private parseFilters(oldFilters: OsFilter<V>[]): OsFilter<V>[] {
+        const newFilterDefs = this.getFilterDefinitions();
+        newFilterDefs.forEach(definition => {
+            const oldDefinition = oldFilters.filter(old => old.property === definition.property);
+            if (!oldDefinition.length) {
+                return;
+            }
+            oldDefinition[0].options.forEach(option => {
+                if (typeof option === `string`) {
+                    return;
+                }
+                const newOption = (
+                    definition.options.filter(newOpt => typeof newOpt !== `string`) as OsFilterOption[]
+                ).filter(newOpt => newOpt.label === option.label);
```

Comment:
Also use `find` here. Also, can you merge the two calls together into one filter like `typeof newOpt !== ``string`` && newOpt.label === option.label`?

Reply:
I tried putting these into the same function, but, even with the check being literally part of the same statement, typescript complains that `newOpt.label` doesn't work if `newOpt` is a string

Reply:
Maybe if you use `typeof newOpt === OsFilterOption` instead? But if not, it's no problem.

Reply:
I think `typeof` only works with basic types